ReactAndroid/src/main/java/com/facebook/react/devsupport/DevServerHelper.java

```diff
@@ -218,8 +218,10 @@ public String getInspectorDeviceUrl() {
   /**
    * @return the host to use when connecting to the bundle server from the host itself.
    */
-  private static String getHostForJSProxy() {
-    return AndroidInfoHelpers.DEVICE_LOCALHOST;
+  private String getHostForJSProxy() {
+    String host = getDebugServerHost();
```

Comment:
You are looking at the wrong method.

The above patch calls `DevServerHelper.getDebugServerHost()`, while you are looking at `DevInternalSettings.getDebugServerHost()`.

DevServerHelper.getDebugServerHost() cannot - ever - return null. Look at its implementation, in GitHub master:

```Java
/**
   * @return the host to use when connecting to the bundle server.
   */
  private String getDebugServerHost() {
    // Check debug server host setting first. If empty try to detect emulator type and use default
    // hostname for those
    String hostFromSettings = mSettings.getDebugServerHost();

    if (!TextUtils.isEmpty(hostFromSettings)) {
      return Assertions.assertNotNull(hostFromSettings);
    }

    String host = AndroidInfoHelpers.getServerHost();

    if (host.equals(AndroidInfoHelpers.DEVICE_LOCALHOST)) {
      FLog.w(
        ReactConstants.TAG,
        "You seem to be running on device. Run 'adb reverse tcp:8081 tcp:8081' " +
          "to forward the debug server's port to the device.");
    }

    return host;
  }
```